src/ring_theory/ideal/basic.lean

```diff
@@ -184,23 +184,16 @@ lemma bot_prime {R : Type*} [ring R] [is_domain R] : (⊥ : ideal R).is_prime :=
 ⟨λ h, one_ne_zero (by rwa [ideal.eq_top_iff_one, submodule.mem_bot] at h),
  λ x y h, mul_eq_zero.mp (by simpa only [submodule.mem_bot] using h)⟩
 
-/-- An ideal is maximal if it is maximal in the collection of proper ideals. -/
-class is_maximal (I : ideal α) : Prop := (out : is_coatom I)
-
-theorem is_maximal_def {I : ideal α} : I.is_maximal ↔ is_coatom I := ⟨λ h, h.1, λ h, ⟨h⟩⟩
-
-theorem is_maximal.ne_top {I : ideal α} (h : I.is_maximal) : I ≠ ⊤ := (is_maximal_def.1 h).1
-
 theorem is_maximal_iff {I : ideal α} : I.is_maximal ↔
   (1:α) ∉ I ∧ ∀ (J : ideal α) x, I ≤ J → x ∉ I → x ∈ J → (1:α) ∈ J :=
-is_maximal_def.trans $ and_congr I.ne_top_iff_one $ forall_congr $ λ J,
+submodule.is_maximal_def.trans $ and_congr I.ne_top_iff_one $ forall_congr $ λ J : ideal α,
 by rw [lt_iff_le_not_le]; exact
  ⟨λ H x h hx₁ hx₂, J.eq_top_iff_one.1 $
     H ⟨h, not_subset.2 ⟨_, hx₂, hx₁⟩⟩,
   λ H ⟨h₁, h₂⟩, let ⟨x, xJ, xI⟩ := not_subset.1 h₂ in
    J.eq_top_iff_one.2 $ H x h₁ xI xJ⟩
 
-theorem is_maximal.eq_of_le {I J : ideal α}
+theorem _root_.submodule.is_maximal.eq_of_le {I J : ideal α}
   (hI : I.is_maximal) (hJ : J ≠ ⊤) (IJ : I ≤ J) : I = J :=
 eq_iff_le_not_lt.2 ⟨IJ, λ h, hJ (hI.1.2 _ h)⟩
```

Comment:
This proof should work just fine about submodules, right? Can you move it to `module/submodule/lattice.lean`?

Reply:
Yes right, thanks. I've moved it to `submodule/lattice.lean`.